Signal/src/ViewControllers/DebugUI/DebugUISessionState.m

```diff
@@ -20,63 +20,112 @@ - (NSString *)name
+                        }]
+    ]];
+#endif
 
-                                           NSData *currentKey = [identityManager identityKeyForRecipientId:recipientId];
-                                           NSMutableData *flippedKey = [NSMutableData new];
-                                           const char *currentKeyBytes = currentKey.bytes;
-                                           for (NSUInteger i = 0; i < currentKey.length; i++) {
-                                               const char xorByte = currentKeyBytes[i] ^ 0xff;
-                                               [flippedKey appendBytes:&xorByte length:1];
-                                           }
-                                           OWSAssert(flippedKey.length == currentKey.length);
-                                           [identityManager saveRemoteIdentity:flippedKey recipientId:recipientId];
-                                       }],
-                       [OWSTableItem itemWithTitle:@"Delete all sessions"
-                                       actionBlock:^{
-                                           dispatch_async([OWSDispatch sessionStoreQueue], ^{
-                                               [[TSStorageManager sharedManager]
-                                                   deleteAllSessionsForContact:thread.contactIdentifier];
-                                           });
-                                       }],
-                       [OWSTableItem itemWithTitle:@"Archive all sessions"
-                                       actionBlock:^{
-                                           dispatch_async([OWSDispatch sessionStoreQueue], ^{
-                                               [[TSStorageManager sharedManager]
-                                                   archiveAllSessionsForContact:thread.contactIdentifier];
-                                           });
-                                       }],
-                       [OWSTableItem itemWithTitle:@"Send session reset"
-                                       actionBlock:^{
-                                           [OWSSessionResetJob runWithContactThread:thread
-                                                                      messageSender:[Environment current].messageSender
-                                                                     storageManager:[TSStorageManager sharedManager]];
-                                       }]
-                   ]];
+    return [OWSTableSection sectionWithTitle:self.name items:items];
 }
 
+#if DEBUG
++ (void)clearSessionAndIdentityStore
+{
+    dispatch_async([OWSDispatch sessionStoreQueue], ^{
+        [[TSStorageManager sharedManager] resetSessionStore];
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [[OWSIdentityManager sharedManager] clearIdentityState];
+        });
+    });
+}
+
++ (void)archiveSessionAndIdentityStore
```

Comment:
Takes a snapshot of session and identity state.

Reply:
Just a naming nit, but "archive" is kind of a reserved word when talking about sessions. I assumed this method did something like the similarly named `archiveAllSessions` method

Can we instead use the term "snapshot" unless we're talking about archiving in the sense of `[session archive]`;

Reply:
Good point, done.